State initialState
State goalState
---
Action[] actions
